src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,95 @@
-import { useState } from 'react';
+import React, { useEffect, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [newTitle, setNewTitle] = useState('');
+  const [newDescription, setNewDescription] = useState('');
+  const [newImgUrl, setNewImgUrl] = useState('');
+  const [newImdbUrl, setNewImdbUrl] = useState('');
+  const [newImdbId, setNewImdbId] = useState('');
+  const [isDisabled, setIsDisabled] = useState(true);
+
+  useEffect(() => {
+    if (newTitle
+      && newImgUrl
+      && newImdbUrl
+      && newImdbId) {
+      setIsDisabled(false);
+    } else {
+      setIsDisabled(true);
+    }
+  });
+
+  const newMovie = {
+    title: newTitle,
+    description: newDescription,
+    imgUrl: newImgUrl,
+    imdbUrl: newImdbUrl,
+    imdbId: newImdbId,
```

Comment:
you can use this key names in your setState variable names and simplify it like this:
```suggestion
    title,
    description,
    imgUrl,
    imdbUrl,
    imdbId,
```